test/generator/example_generator.cpp

```diff
@@ -81,13 +81,13 @@ class Example : public Halide::Generator<Example> {
             .bound(c, 0, channels)
             .reorder(c, x, y)
             .unroll(c);
-        if (vectorize) {
-            // Note that we can use the Generator method natural_vector_size()
-            // here; this produces the width of the SIMD vector being targeted
-            // divided by the width of the data type.
-            output
-                .vectorize(x, natural_vector_size(output.type()));
-        }
+        // Note that we can use the Generator method natural_vector_size()
+        // here; this produces the width of the SIMD vector being targeted
+        // divided by the width of the data type.
+        // Use '1' as the 'no vectorize' case.
+        // TODO: this is awkward, there should be a more idiomatic way.
+        output
+            .vectorize(x, select(vectorize, natural_vector_size(output.type()), 1));
```

Comment:
I think that in practice, I would tend to reserve usage of this for things that need to be deferred, like loop levels for line buffer storage. Things like vectorizing/parallelizing via schedule params seems a bit contrived, I think most code in practice would base those decisions on the target. So, while I think we should support this use case, I don't see the combinatoric specialize issue being a big problem. Of course, ScheduleParam<bool> should still work.